Doc updates for 3.0 release (#2568)
zarr-developers/zarr-python: docs/user-guide/storage.rst

The :class:`zarr.storage.LocalStore` stores data in a nested set of directories on a local
filesystem.

.. ipython:: python

   store = zarr.storage.LocalStore("data/foo/bar", read_only=True)
   zarr.open(store=store, mode='r')

Zip Store
~~~~~~~~~

The :class:`zarr.storage.ZipStore` stores the contents of a Zarr hierarchy in a single

> Zip file. The `Zip Store specification_` is currently in draft form.

Reviewer: This needs a link in the form of `Zip Store specification <insert link>`_

.. ipython:: python

   store = zarr.storage.ZipStore("data.zip", mode="w")
   zarr.open(store=store, shape=(2,))

Remote Store
~~~~~~~~~~~~

The :class:`zarr.storage.FsspecStore` stores the contents of a Zarr hierarchy in following the same
logical layout as the ``LocalStore``, except the store is assumed to be on a remote storage system
such as cloud object storage (e.g. AWS S3, Google Cloud Storage, Azure Blob Store). The

> :class:`zarr.storage.FsspecStore` is backed by `Fsspec_` and can support any Fsspec backend

Reviewer: ```suggestion
:class:`zarr.storage.FsspecStore` is backed by `Fsspec <https://filesystem-spec.readthedocs.io/en/latest/>`_ and can support any Fsspec backend
```

that implements the `AbstractFileSystem` API. ``storage_options`` can be used to configure
the Fsspec backend.

.. ipython:: python

   store = zarr.storage.FsspecStore.from_url(
      "s3://noaa-nwm-retro-v2-zarr-pds",
      read_only=True,
      storage_options={"anon": True}
   )
   zarr.open_group(store=store, mode='r')

Memory Store
~~~~~~~~~~~~

> The :class:`zarr.storage.FsspecStore` a in-memory store that allows for serialization of
> Zarr data (metadata and chunks) to a dictionary.

Reviewer: ```suggestion
The :class:`zarr.storage.MemoryStore` allows for serialization of Zarr data (metadata and chunks) to an im-memory dictionary.
```
